mos_tests/rabbitmq_oslo/test_restarts.py

```diff
@@ -0,0 +1,156 @@
+           "pip install -r requirements.txt -r test-requirements.txt && "
+           "dpkg -i {pkg} || "
+           "apt-get -f install -y".format(root_path=kwargs['root_path'],
+                                          repo=kwargs['repo'],
+                                          repo_path=kwargs['repo_path'],
+                                          pkg=kwargs['pkg']))
+    logger.debug('Install "oslo.messaging-check-tool" on controller')
+    return remote.check_call(cmd)
+
+
+def configure_oslomessagingchecktool(remote, ctrl_ips, **kwargs):
+    """Write configuration file on controller"""
+    # Create config file for "oslo.messaging-check-tool"
+    configs = (
+        "[DEFAULT]\n"
+        "debug=true\n"
+        "[oslo_messaging_rabbit]\n"
+        "rabbit_hosts = {rabbit_hosts}\n"
+        "rabbit_userid = {nova_user}\n"
+        "rabbit_password = {nova_pass}\n".format(
+            rabbit_hosts=":5673, ".join(ctrl_ips),
+            nova_user=kwargs['nova_user'],
+            nova_pass=kwargs['nova_pass']))
+    # Write config to file on controller
+    logger.debug('Write "oslo.messaging-check-tool" config file to controller')
+    with remote.open(kwargs['conf_file_path'], 'w') as f:
+        f.write(configs)
+
+
+def restart_rabbitmq_serv(env, remote=None, restart_all=False):
+    """Restart rabbitmq-server service on one or all controllers
+    :param env: Environment
+    :param remote: SSH connection point to controller.
+        Leave empty if you want to restart service on all controllers.
+    :param restart_all: Set True to restart on service on all controllers.
+    :return: None
+    """
+    restart_cmd = 'service rabbitmq-server restart'
+    # if certain controller not set --> restart rabbitmq on all controllers
+    if remote is None:
+        restart_all = True
+    # restart on one controller
+    if remote is not None:
+        logger.debug('Restart RabbinMQ server on one controller')
+        remote.check_call(restart_cmd)
+    # restart on all controllers
+    if restart_all is True:
+        controllers = env.get_nodes_by_role('controller')
+        logger.debug('Restart RabbinMQ server on all controllers')
+        for controller in controllers:
+            with controller.ssh() as remote:
+                remote.check_call(restart_cmd)
+
+
+@pytest.mark.undestructive
+@pytest.mark.testrail_id('838284')
+def test_load_messages_and_restart_one_controller(env):
+    """Load 10000 messages to RabbitMQ cluster and restart RabbitMQ
+    on one controller.
+
+    Actions:
+    1. Install "oslo.messaging-check-tool" on controller;
+    2. Prepare config file for it;
+    3. Generate 10000 messages to RabbitMQ cluster;
+    4. Restart RabbitMQ-server on one controller;
+    5. Consume messages;
+    6. Check that number of generated and consumed messages is equal.
+    """
+    controllers = env.get_nodes_by_role('controller')
+    controller = controllers[(randint(0, len(controllers) - 1))]
```

Comment:
Actually, no, it is not required to have random controllers.
Just in order to have less constants in test I'd like to have random choice every time.